icu4j/main/classes/core/src/com/ibm/icu/impl/number/MixedUnitLongNameHandler.java

```diff
@@ -0,0 +1,197 @@
+     */
+    public static MixedUnitLongNameHandler forMeasureUnit(ULocale locale, MeasureUnit mixedUnit,
+                                                          NumberFormatter.UnitWidth width, PluralRules rules,
+                                                          MicroPropsGenerator parent) {
+        assert (mixedUnit.getComplexity() == MeasureUnit.Complexity.MIXED);
+
+        MixedUnitLongNameHandler result = new MixedUnitLongNameHandler(rules, parent);
+        List<MeasureUnit> individualUnits = mixedUnit.splitToSingleUnits();
+
+        result.fMixedUnitData = new ArrayList<>();
+        for (int i = 0; i < individualUnits.size(); i++) {
+            // Grab data for each of the components.
+            String[] unitData = new String[LongNameHandler.ARRAY_LENGTH];
+            LongNameHandler.getMeasureData(locale, individualUnits.get(i), width, unitData);
+            result.fMixedUnitData.add(unitData);
+        }
+
+        ListFormatter.Width listWidth = ListFormatter.Width.SHORT;
+        if (width == NumberFormatter.UnitWidth.NARROW) {
+            listWidth = ListFormatter.Width.NARROW;
+        } else if (width == NumberFormatter.UnitWidth.FULL_NAME) {
+            // This might be the same as SHORT in most languages:
+            listWidth = ListFormatter.Width.WIDE;
+        }
+
+        result.fListFormatter = ListFormatter.getInstance(locale, ListFormatter.Type.UNITS, listWidth);
+
+
+        // We need a localised NumberFormatter for the integers of the bigger units
+        // (providing Arabic numerals, for example).
+        result.fIntegerFormatter = NumberFormatter.withLocale(locale);
+
+        return result;
+    }
+
+    /**
+     * Produces a plural-appropriate Modifier for a mixed unit: `quantity` is
+     * taken as the final smallest unit, while the larger unit values must be
+     * provided by `micros.mixedMeasures`, micros being the MicroProps instance
+     * returned by the parent.
+     *
+     * This function must not be called if this instance has no parent: call
+     * processQuantityWithMicros() instead.
+     */
+    @Override
+    public MicroProps processQuantity(DecimalQuantity quantity) {
+        assert (fMixedUnitData.size() > 1);
+        MicroProps micros;
+        micros = parent.processQuantity(quantity);
+        micros.modOuter = getMixedUnitModifier(quantity, micros);
+        return micros;
+    }
+
+    /**
+     * Produces a plural-appropriate Modifier for a mixed unit: `quantity` is
+     * taken as the final smallest unit, while the larger unit values must be
+     * provided via `micros.mixedMeasures`.
+     */
+    public MicroProps processQuantityWithMicros(DecimalQuantity quantity, MicroProps micros) {
+        assert (fMixedUnitData.size() > 1);
+        micros.modOuter = getMixedUnitModifier(quantity, micros);
+        return micros;
+    }
+
+    /**
+     * Required for ModifierStore. And ModifierStore is required by
```

Comment:
If this is "Required for ModifierStore", why is it not `@Override` ?

Reply:
done